spec/models/result_spec.rb

```diff
@@ -199,13 +199,15 @@
                   context 'when marks are created for this incomplete result' do
                     let!(:incomp_result) { results[2] }
                     let!(:prev_subtotal) { incomp_result.get_subtotal }
+                    let!(:flex_criteria_first) { create(:flexible_criterion, assignment: assignment) }
+                    let!(:flex_criteria_second) { create(:flexible_criterion, max_mark: 2.0, assignment: assignment) }
                     before do
-                      create(:flexible_mark, result: incomp_result, mark: 1)
-                      create(:flexible_mark, result: incomp_result, mark: 1)
+                      create(:flexible_mark, result: incomp_result, mark: 1, markable: flex_criteria_first)
```

Comment:
@WChrisK I'm wondering if you tried just doing `assignment: assignment` right here. It might not work because it's really a nested association, but I thought it would be worth a try.